### PR TITLE
Various tidying and refactoring

### PR DESCRIPTION
**Pull request by alexjfisher**
* Stop using relative class/variable names
* Remove duplicate resource relationships
* Use `facts` hash instead of topscope variables
* Don't use `Variant[Undef, Array]`

### CODE REVIEW

File: manifests/install.pp [hunk 1]
```diff
@@ -49,19 +49,19 @@
...
+  if $facts['osfamily'] == 'windows' {
     # required to install telegraf on windows
     require chocolatey
 
     # package install
-    package { $::telegraf::package_name:
-      ensure          => $::telegraf::ensure,
+    package { $telegraf::package_name:
+      ensure          => $telegraf::ensure,
       provider        => chocolatey,
```
Comment (ekohl) [inline manifests/install.pp:59]: Should this be quoted?

**Review by ekohl — COMMENTED** [review]
Looks good

File: manifests/install.pp (hunk 1 above)
Comment (alexjfisher) [reply manifests/install.pp:59]: Style-wise? Not sure. It is valid though as is. https://puppet.com/docs/puppet/5.2/lang_data_string.html#bare-words

Comment (alexjfisher) [reply manifests/install.pp:59]: @ekohl Good to merge?

**Review by ekohl — APPROVED** [review]

**Pull request merged**